apps/server/src/shared/repo/userloginmigration/user-login-migration.repo.ts

```diff
@@ -0,0 +1,62 @@
+import { EntityName } from '@mikro-orm/core';
+import { EntityManager } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { EntityId, School, System, UserLoginMigrationDO } from '@shared/domain';
+import { IUserLoginMigration, UserLoginMigration } from '@shared/domain/entity/user-login-migration.entity';
+import { Logger } from '@src/core/logger';
+import { BaseDORepo } from '../base.do.repo';
+
+@Injectable()
+export class UserLoginMigrationRepo extends BaseDORepo<UserLoginMigrationDO, UserLoginMigration, IUserLoginMigration> {
+	constructor(protected readonly _em: EntityManager, protected readonly logger: Logger) {
+		super(_em, logger);
+	}
+
+	get entityName(): EntityName<UserLoginMigration> {
+		return UserLoginMigration;
+	}
+
+	entityFactory(props: IUserLoginMigration): UserLoginMigration {
+		return new UserLoginMigration(props);
+	}
+
+	async findBySchoolId(schoolId: EntityId): Promise<UserLoginMigrationDO | null> {
+		const userLoginMigration: UserLoginMigration | null = await this._em.findOne(UserLoginMigration, {
+			school: schoolId,
+		});
+
+		const userLoginMigrationDO: UserLoginMigrationDO | null = userLoginMigration
+			? this.mapEntityToDO(userLoginMigration)
+			: null;
+		return userLoginMigrationDO;
```

Comment:
personally, I find it easier to read with simple iff, and no need to specify type null
moreover, not so sure if defining types for each constant is required, because the type can be inferred, but ok.

if (userLoginMigration) {
  const userLoginMigrationDO =  this.mapEntityToDO(userLoginMigration);
  return userLoginMigrationDO;
}

return null;